tests/MeltorTests.php

```diff
@@ -0,0 +1,320 @@
+        $this->assertEquals(app('meltor')->config('subkey.closure'), 'closureResult');
+    }
+
+    /**
+     * @test
+     * @return void
+     */
+    public function canStripDatabaseName(): void
+    {
+        $this->assertEquals(app('meltor')->stripDatabaseName('database_name/table_name'), 'table_name');
+        $this->assertEquals(app('meltor')->stripDatabaseName('table_name'), 'table_name');
+    }
+
+    /**
+     * @test
+     * @return void
+     */
+    public function canGetDatabaseConfigOfProject(): void
+    {
+        $this->assertTrue(array_key_exists('driver', app('meltor')->getDatabaseConfig()));
+    }
+
+
+    /**
+     * @test
+     * @return void
+     */
+    public function canGetDatabaseStructure(): void
+    {
+        $result = app('meltor')->getDatabaseStructure('meltor', app('db')->connection('information_schema'));
+
+        $this->assertArrayHasKey('migrations', $result);
+    }
+
+    /**
+     * @test
+     * @return void
+     */
+    public function canGetUniqueKeys(): void
+    {
+        $result = app('meltor')->getUniqueKeys('meltor', app('db')->connection('information_schema'));
+
+        $this->assertArrayHasKey('meltor_all_types_test', $result);
+        $this->assertEquals('unique_one_field', $result['meltor_all_types_test'][0]->INDEX_NAME);
+    }
+
+    /**
+     * @test
+     * @return void
+     */
+    public function canGetForeignKeys(): void
+    {
+        $result = app('meltor')->getForeignKeys('meltor', app('db')->connection('information_schema'));
+
+        $this->assertArrayHasKey('meltor_all_types_test', $result);
+        $this->assertEquals('meltor_foreign_id', $result['meltor_all_types_test'][0]->COLUMN_NAME);
+    }
+
+    /**
+     * @test
+     * @return void
+     */
+    public function canGetDataTypeFromColumn(): void
+    {
+        $validColumn           = $this->getDummyColumn();
+        $validBoolColumn       = $this->getDummyColumn('int', 'tinyint(1)');
+        $invalidFileTypeColumn = $this->getDummyColumn('int', 'int', 'unknown type');
+
+        // Special boolean handling
+        $this->assertEquals('boolean', $this->runProtectedMethod('getDataType', [$validBoolColumn]));
+
+        // Regular handling
+        $this->assertEquals('int', $this->runProtectedMethod('getDataType', [$validColumn]));
+
+        // Invalid MySQL DATA_TYPE, ignore problems mode
+        $this->assertNull($this->runProtectedMethod('getDataType', [$invalidFileTypeColumn, true]));
+
+        // Invalid MySQL DATA_TYPE
+        $this->expectException(Exception::class);
+        $this->runProtectedMethod('getDataType', [$invalidFileTypeColumn]);
+    }
+
+    /**
+     * @test
+     * @return void
+     */
+    public function canGetExtraFromColumn(): void
+    {
+        $validColumn              = $this->getDummyColumn();
+        $validAutoIncrementColumn = $this->getDummyColumn('int', 'int', 'int', 'auto_increment');
```

Comment:
```suggestion
        $validAutoIncrementColumn = $this->getDummyColumn(mysqlType: 'int', columnType: 'int', dataType: 'int', extra: 'auto_increment');
```
This makes it a little more readable.